src/models/LightningBaseModel.py

```diff
@@ -90,58 +89,81 @@ def test_step(self, batch, batch_idx, *args, **kwargs):
 
     def _do_step(self, images, labels, label_names, step, log_images=False):
 
-        predictions = self(images)
+        class_log_probabilities = self(images)
+        class_probabilities = F.softmax(class_log_probabilities, dim=1).detach().cpu()
+        labels_est = class_log_probabilities.argmax(dim=-1).detach().cpu()
+        targets = labels.view(-1).to(torch.int).detach().cpu()
 
-        loss = self.loss_func(predictions, labels.view(-1).long())
-        accuracy = self.accuracy_func(F.softmax(predictions, dim=1).detach().cpu(), labels.to(torch.int).detach().cpu())
-
-        labels_est = predictions.argmax(dim=-1).detach().cpu()
-        targets = labels.to(torch.int).detach().cpu()
+        loss = self.loss_func(class_log_probabilities, labels.view(-1).long())
+        accuracy = self.accuracy_func(class_probabilities, targets)
 
         # lets log some values for inspection (for example in tensorboard):
         self.log(f"NLL {step}", loss)
         self.log(f"Accuracy {step}", accuracy)
 
         if self.log_confusion_matrices:
-            self._update_confusion_matrix(step, targets, labels_est)
+            self._update_accuracy_matrices(step, targets, labels_est)
 
         if log_images:
             self.log_images(images, label_names)
 
         return loss, accuracy
 
-    def _update_confusion_matrix(self, datagroup, labels_true, labels_est):
+    def _update_accuracy_matrices(self, datagroup, labels_true, labels_est):
         # sum over batch to update confusion matrix
         n_classes = len(self.class_labels)
         idx = labels_true + n_classes * labels_est
         counts = np.bincount(idx.reshape(-1), minlength=n_classes ** 2)
-        self.CM[datagroup] += counts.reshape((n_classes, n_classes))
+        self.confusion_matrix[datagroup] += counts.reshape((n_classes, n_classes))
 
-    def _init_confusion_matrices(self):
+    def _init_accuracy_matrices(self):
+        n = len(self.class_labels)
         for datagroup in ['Validation', 'Training', 'Testing']:
-            self.CM[datagroup] = np.zeros((len(self.class_labels), len(self.class_labels)), dtype=np.int64)
+            self.confusion_matrix[datagroup] = np.zeros((n, n), dtype=np.int64)
+
+    def _log_accuracy_matrices(self, datagroup):
+        cm = self.confusion_matrix[datagroup]
+        n = cm.sum()
+        with np.errstate(divide='ignore', invalid='ignore'):
```

Comment:
is this safe? Is accuracy defined if the devision fails? Maybe define accuracy and conditional_probabilities before and set them to 0 just in case, so that the code does not fail when trying to log these later?

Reply:
changed nans to zeros